src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -3244,6 +3244,18 @@ void MacroAssembler::vpmullw(XMMRegister dst, XMMRegister nds, Address src, int
   Assembler::vpmullw(dst, nds, src, vector_len);
 }
 
+void MacroAssembler::vpmulld(XMMRegister dst, XMMRegister nds, AddressLiteral src, int vector_len) {
+  // Used in sign-bit flipping with aligned address.
+  bool aligned_adr = (((intptr_t)src.target() & 15) == 0);
```

Comment:
This is an AVX instruction. So alignment is not required. The assert need to only check for UseAVX>0.